### PR TITLE
Bump golang version to 1.21

### PR DESCRIPTION
**Pull request by spotlesstofu**
**- Description of the problem which is fixed/What is the use case**

**- What I did**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Review by gkurz — APPROVED** [review]
/lgtm

Thanks @spotlesstofu !

**Comment by bpradipt** [comment]
@spotlesstofu shouldn't this be targeted for devel branch ?

**Comment by gkurz** [comment]
> @spotlesstofu shouldn't this be targeted for devel branch ?

Indeed I overlooked it was targetting a dummy branch... @spotlesstofu is the PR ok to go to `devel` ?

**Comment by spotlesstofu** [comment]
@gkurz it is OK now. The bot tells it needs rebase, but it shouldn't. I changed the target branch and rebased already.

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Use OpenShift golang builder image
 # These images needs to be synced with the images in the Makefile.
-ARG BUILDER_IMAGE=${BUILDER_IMAGE:-registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.19-openshift-4.13}
+ARG BUILDER_IMAGE=${BUILDER_IMAGE:-registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.21-openshift-4.13}
```
Comment (bpradipt) [inline Dockerfile]: does it makes sense to use the following images ?
BUILDER_IMAGE=registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.21-openshift-4.16
TARGET_IMAGE=registry.ci.openshift.org/ocp/4.16:base-rhel9

Ref: https://github.com/openshift/release/blob/077ae802a5461f63a34ba5157cc2bc2783db7219/ci-operator/config/openshift/oc/openshift-oc-release-4.16.yaml

Comment (gkurz) [reply Dockerfile]: These are  likely the images @spotlesstofu is looking for. Thanks @bpradipt !

**Review by bpradipt — APPROVED** [review]
/lgtm

**Comment by gkurz** [comment]
Thanks @spotlesstofu !

We don't formally ask  for it but it is generally good practice to add some extra info in the commit message and a S-o-b tag.

e.g.

```
Bump golang version to 1.21

Downstream OSC was already bumped to use golang 1.21. We should at least use
that version for any custom builds from now on.

Next version of OCP we'll need to support is 4.16. Their CI operator relies on golang
1.21 builder image at [1]. Let's use this same image.

[1] https://github.com/openshift/release/blob/077ae802a5461f63a34ba5157cc2bc2783db7219/ci-operator/config/openshift/oc/openshift-oc-release-4.16.yaml#L25

Signed-off-by: Camilla Conte <cconte@redhat.com>
```

**Comment by spotlesstofu** [comment]
Thanks @gkurz, just updated with your commit message.

**Pull request merged**